src/rhsm/connection.py

```diff
@@ -83,25 +83,28 @@ def normalized_host(host: str) -> str:
         return host
 
 
-def drift_check(utc_time_string: str, hours: int = 1) -> bool:
-    """
-    Takes in a RFC 1123 date and returns True if the current time
-    is greater than the supplied number of hours
+def drift_check(timestamp: str, hours: int = 1) -> bool:
+    """Verify a timestamp is not too much different from local clock.
+
+    :param timestamp: A timezone-unaware timestamp in RFC 1123 format.
+    :param hours: Allowed time drift in hours.
+    :returns: True if the timestamp is more than 'hours' off from local time.
     """
-    drift = False
-    if utc_time_string:
-        try:
-            # This may have a timezone (utc)
-            utc_datetime = dateutil.parser.parse(utc_time_string)
-            # This should not have a timezone, but we know it will be utc.
-            # We need our timezones to match in order to compare
-            local_datetime = datetime.datetime.utcnow().replace(tzinfo=utc_datetime.tzinfo)
-            delta = datetime.timedelta(hours=hours)
-            drift = abs((utc_datetime - local_datetime)) > delta
-        except Exception as e:
-            log.error(e)
-
-    return drift
+    if not timestamp:
+        return False
+
+    # The time has format of 'Fri, 12 Jan 2024 08:10:46 GMT'
+    timestamp: datetime.datetime = dateutil.parser.parse(timestamp)
+    if timestamp.tzinfo.tzname(timestamp) != "UTC":
+        log.warning(f"Expected UTC timestamp, got '{timestamp}', drift check may be off.")
+    # dateutil has its own tzinfo object representing UTC
+    timestamp = timestamp.replace(tzinfo=datetime.timezone.utc)
+
+    now = datetime.datetime.now(datetime.timezone.utc).replace(microsecond=0)
+    delta = datetime.timedelta(hours=hours)
+
+    drift: datetime.timedelta = abs(timestamp - now)
+    return drift > delta
```

Comment:
We could also return the value of drift and print it in warning message then.

Reply:
Good idea.